File: pytools/data/data.code.golang.server/req_role_login.go
package proto

import (
	"packet"
)

type ReqRoleLogin struct {
	uid                      uint32
	uuid                     uint32
	sid                      uint16
	cid                      uint16
	loginTime                uint32
	pwd                      string
	relink                   uint8
	debug                    uint8
	os                       string
	version                  string
}

func ReqRoleLoginDecode(pack *packet.Packet) *ReqRoleLogin {
	reqRoleLogin := &ReqRoleLogin{}

	reqRoleLogin.uid = pack.ReadUint32()
	reqRoleLogin.uuid = pack.ReadUint32()
	reqRoleLogin.sid = pack.ReadUint16()
	reqRoleLogin.cid = pack.ReadUint16()
	reqRoleLogin.loginTime = pack.ReadUint32()
	reqRoleLogin.pwd = pack.ReadString()
	reqRoleLogin.relink = pack.ReadUint8()
	reqRoleLogin.debug = pack.ReadUint8()
	reqRoleLogin.os = pack.ReadString()
	reqRoleLogin.version = pack.ReadString()
	return reqRoleLogin
}

func (this *ReqRoleLogin) GetUid() uint32 {
	return this.uid
}

func (this *ReqRoleLogin) GetUuid() uint32 {
	return this.uuid
}

func (this *ReqRoleLogin) GetSid() uint16 {
	return this.sid
}

func (this *ReqRoleLogin) GetCid() uint16 {
	return this.cid
}

func (this *ReqRoleLogin) GetLoginTime() uint32 {
	return this.loginTime
}

func (this *ReqRoleLogin) GetPwd() string {
	return this.pwd
}

func (this *ReqRoleLogin) GetRelink() uint8 {
	return this.relink
}

func (this *ReqRoleLogin) GetDebug() uint8 {
	return this.debug
}

func (this *ReqRoleLogin) GetOs() string {
	return this.os
}

func (this *ReqRoleLogin) GetVersion() string {
	return this.version
}

File: pytools/data/data.code.cpp.server/ReqRoleLogin.cpp
#ifndef _REQ_ROLE_LOGIN_
#define _REQ_ROLE_LOGIN_

#include <list>

#include <pb_type.hpp>
#include <Packet.hpp>

#include <Msg.hpp>



class ReqRoleLogin
{
private:
	U32 m_uid;
	U32 m_uuid;
	U16 m_sid;
	U16 m_cid;
	U32 m_login_time;
	String m_pwd;
	U8 m_relink;
	U8 m_debug;
	String m_os;
	String m_version;


public:
	ReqRoleLogin(Packet* packet)
	{
		m_uid = packet->ReadU32();
		m_uuid = packet->ReadU32();
		m_sid = packet->ReadU16();
		m_cid = packet->ReadU16();
		m_login_time = packet->ReadU32();
		m_pwd = packet->ReadString();
		m_relink = packet->ReadU8();
		m_debug = packet->ReadU8();
		m_os = packet->ReadString();
		m_version = packet->ReadString();
	}


	U32 GetUid()
	{
		return m_uid;
	}

	U32 GetUuid()
	{
		return m_uuid;
	}

	U16 GetSid()
	{
		return m_sid;
	}

	U16 GetCid()
	{
		return m_cid;
	}

	U32 GetLoginTime()
	{
		return m_login_time;
	}

	String GetPwd()
	{
		return m_pwd;
	}

	U8 GetRelink()
	{
		return m_relink;
	}

	U8 GetDebug()
	{
		return m_debug;
	}

	String GetOs()
	{
		return m_os;
	}

	String GetVersion()
	{
		return m_version;
	}

};

#endif
